src/hooks/api/events.ts

```diff
@@ -0,0 +1,304 @@
+} from './authenticated-query';
+import type { Headers } from './types/Headers';
+import type { ServerSideArguments } from './types/ServerSideArguments';
+import type { SortOptions } from './types/SortOptions';
+
+type HeadersAndQueryData = {
+  headers: Headers;
+} & { [x: string]: string };
+
+type GetEventsToModerateArguments = HeadersAndQueryData;
+
+const getEventsToModerate = async ({
+  headers,
+  ...queryData
+}: GetEventsToModerateArguments) => {
+  const res = await fetchFromApi({
+    path: '/events/',
+    searchParams: {
+      ...queryData,
+      availableFrom: formatDate(new Date()),
+    },
+    options: {
+      headers,
+    },
+  });
+  if (isErrorObject(res)) {
+    // eslint-disable-next-line no-console
+    return console.error(res);
+  }
+  return await res.json();
+};
+
+const useGetEventsToModerate = (
+  searchQuery: string,
+  configuration: UseQueryOptions = {},
+) =>
+  useAuthenticatedQuery({
+    queryKey: ['events'],
+    queryFn: getEventsToModerate,
+    queryArguments: {
+      q: searchQuery,
+      audienceType: 'everyone',
+      availableTo: '*',
+      limit: 1,
+      start: 0,
+      workflowStatus: 'READY_FOR_VALIDATION',
+    },
+    enabled: !!searchQuery,
+    ...configuration,
+  });
+
+type GetEventByIdArguments = {
+  headers: Headers;
+  id: string;
+};
+
+const getEventById = async ({ headers, id }: GetEventByIdArguments) => {
+  const res = await fetchFromApi({
+    path: `/event/${id.toString()}`,
+    options: {
+      headers,
+    },
+  });
+  if (isErrorObject(res)) {
+    // eslint-disable-next-line no-console
+    return console.error(res);
+  }
+  return await res.json();
+};
+
+type UseGetEventsByIdArguments = ServerSideArguments & {
+  id: string;
+};
+
+const useGetEventById = (
+  { req, queryClient, id }: UseGetEventsByIdArguments,
+  configuration: UseQueryOptions = {},
+) =>
+  useAuthenticatedQuery({
+    req,
+    queryClient,
+    queryKey: ['events'],
+    queryFn: getEventById,
+    queryArguments: { id },
+    enabled: !!id,
+    ...configuration,
+  });
+
+type UseGetEventsByIdsArguments = ServerSideArguments & {
+  ids: string[];
+};
+
+const useGetEventsByIds = ({
+  req,
+  queryClient,
+  ids = [],
+}: UseGetEventsByIdsArguments) => {
+  const options = ids.map((id) => ({
+    queryKey: ['events'],
+    queryFn: getEventById,
+    queryArguments: { id },
+    enabled: !!id,
+  }));
+
+  return useAuthenticatedQueries({ req, queryClient, options });
+};
+
+type GetEventsByCreatorArguments = HeadersAndQueryData;
+
+const getEventsByCreator = async ({
+  headers,
+  ...queryData
+}: GetEventsByCreatorArguments) => {
+  const res = await fetchFromApi({
+    path: '/events/',
+    searchParams: {
+      ...queryData,
+    },
+    options: {
+      headers,
+    },
+  });
+  if (isErrorObject(res)) {
+    // eslint-disable-next-line no-console
+    return console.error(res);
+  }
+  return await res.json();
+};
+
+type UseGetEventsByCreatorArguments = {
```

Comment:
I wonder if we need to define a type `UseGetEventsByCreatorArguments` for the arguments of a method. I think parameters with types is perfectly fine?

Reply:
The properties creatorid, start, limit, ... live in a object, not as seperate arguments. That's why I can't just say:

```javascript
const useGetEventsByCreator = (
  {
    creatorId: string,
    start: number = 0,
    limit: number = 50,
    sortOptions: SortOptions = { field: 'modified', order: 'desc' },
  }: UseGetEventsByCreatorArguments,
  configuration: UseQueryOptions = {},
) =>
```

In javascript this is seen as use this property by another name

Reply:
Ok, I think this is indeed very easy in JavaScript because calling this method can be done by creating the parameter object on the fly like this:
```
useGetEventsByCreator({
  creatorId: 1,
  start: 0,
  limit: 50,
  sortOptions: {
    field: 'modified',
    order: 'desc'
  }
})
```